### ISSUE TITLE
Ability to disable and detach ProgressBar

### ISSUE THREAD

**Issue by mil-ad** [pytorch/ignite#1303]
In some cased I'd like to disable all `print`s in my training script based on a condition without actually commenting them out or adding `if`s around them. An example of this is when I'm using [ray](http://ray.io/) for parallel training. It'd be useful if we could add a bit state to `ProgressBar` and add `enable()` and `disable()` methods. 

I can inherit `ProgressBar` and make `log_message()` do nothing but it's not obvious to me how to do the opposite of `ProgressBar.attach`.

**Comment by vfdev-5** [comment]
@mi-lad thanks for the feature request. Could you please provide a bit more details of how would you like to see it in terms of API. Currently, I'm doing `if` clause if it is inappropriate to have tqdm logging:
```python
trainer = ...

if condition:
    ProgressBar().attach(trainer)

trainer.run(...)
```
If we imagine a `ProgressBar.detach()` method that detaches progressbar handlers from the engine, how would you use it ?

**Comment by mi-lad** [comment]
Thanks for the quick reply @vfdev-5 . The external API I'm imagining is simply `enable()` and `disable()` methods for `ProgressBar`  that would internally enable/disable both `log_message` _and_ the attached trainer.

I guess one complexity is handling this in the middle of an epoch.

**Comment by vfdev-5** [comment]
@mi-lad seems like you would like something global stdout enable/disable right ?
For example
```python
from ignite.engine import Engine, Events
from ignite.contrib.handlers import ProgressBar

import os
import sys
import time

trainer = Engine(lambda e, b: time.sleep(0.2))

pbar = ProgressBar()
pbar.attach(trainer)

outstream = sys.stdout
errstream = sys.stderr


@trainer.on(Events.ITERATION_STARTED(once=10))
def disable_std_out_err():
    f = open(os.devnull, "w")
    sys.stdout = sys.stderr = f
    
    
@trainer.on(Events.ITERATION_STARTED(once=30))
def enable_std_out_err():
    sys.stdout = outstream
    sys.stderr = errstream

@trainer.on(Events.ITERATION_STARTED)
def log_iter():
    pbar.log_message("{}".format(trainer.state.iteration))

data = range(50)

trainer.run(data)
```

This stops printing from 10-th iteration until 30-th. There is something to do with tty for tqdm's bar, not sure yet how to do that.

**Comment by vfdev-5** [comment]
@mil-ad I close this issue as solved, let me know if you need more support on that.
